api/services/member.services.ts

```diff
@@ -1,60 +1,114 @@
-import { MemberCollectionValidation } from "../validations";
-import { MemberRepository } from "../models";
+import { MemberCollectionValidation, MemberValidation } from "../validations";
+import { MemberRepository, ProductRepository } from "../models";
 import { CreationMember, Member, MemberSchema } from "../types";
+import mongoose from "mongoose";
 
 export class MembersServices {
   static async getAll() {
     return await MemberRepository.find();
   }
 
-  static async getOne(identifier: MemberSchema["_id"]) {
-    const Member = await MemberRepository.findOne({
-      identifier,
-    }).exec();
+  static async getById(id: MemberSchema["_id"]) {
+    const Member = await MemberRepository.findById(id);
 
     if (!Member) {
-      throw new Error("Team member not found");
+      throw new Error("The member was not found");
     }
 
     return Member;
   }
 
-  static async getById(_id: MemberSchema["_id"]) {
-    const Member = await MemberRepository.findById(_id);
+  static async create(data: CreationMember) {
+    MemberValidation.parse(data);
+    return await MemberRepository.create(data);
+  }
 
-    if (!Member) {
-      throw new Error("Team Member not found.");
+  static async assignProductToMember({
+    productId,
+    memberId,
+  }: {
+    productId: string;
+    memberId: string;
+  }) {
+    const member = await MemberRepository.findById(memberId);
+    const product = await ProductRepository.findById(productId);
+
+    if (!member || !product) {
+      throw new Error("Member or product not found");
     }
 
-    return Member;
-  }
+    member.products.push(product);
 
-  static async create(data: CreationMember) {
-    return await MemberRepository.create(data);
+    await member.save();
+
+    await product.remove();
   }
 
+  static async assignManyProductsToMember({
+    memberId,
+    productsIds,
+  }: {
+    memberId: string;
+    productsIds: string[];
+  }) {
+    const session = await mongoose.startSession();
+    session.startTransaction();
+
+    try {
+      const member = await MemberRepository.findById(memberId).session(session);
+
+      if (!member) {
+        throw new Error("Member not found");
+      }
+
```

Comment:
`const result = await Promise.all(productsIds.map(productId => ProductRepository.findById(productId).session(session)))`

Reply:
1. One query to retrieve (and ideally delete) all products from product array.
   a. `const products = await ProductRepository.getAllProductsByIds(productsIds).session(session)`
2. One query to add all retrieved products to member.